extensions/wikia/ContributionAppreciation/ContributionAppreciationController.class.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+class ContributionAppreciationController extends WikiaController {
+
+	public function appreciate() {
+		global $wgUser;
+
+		if ( $this->request->wasPosted() && $wgUser->matchEditToken( $this->getVal( 'token' ) ) ) {
+			//TODO: do something with apprecation
+			$this->response->setFormat( WikiaResponse::FORMAT_JSON );
+			$this->setVal( 'user', $wgUser->getName() );
+			$this->setVal( 'for', Revision::newFromId( $this->request->getInt( 'revision' ) )->getUserText() );
+		}
+	}
+
+	public static function onDiffHeader( DifferenceEngine $diffPage, $oldRev, Revision $newRev ) {
+		Wikia::addAssetsToOutput( 'contribution_appreciation_js' );
+		$diffPage->getOutput()->addHTML( self::getAppreciationLink( $newRev->getId() ) );
```

Comment:
Could we make use of the fact that we have available here both old and new revision and pass to link old revision id as well? Its needed for creating a diff page link ie. in email.